### ISSUE TITLE
Artifacts with invalid version strings can be `put` but not `got`

### ISSUE THREAD

**Issue by kmdouglass** [cf-platform-eng/concourse-pypi-resource#30]
Thanks for taking the time create and maintain this resource. I appreciate it.

I noticed an inconsistent behavior when uploading the following artifact to a private PyPI server backed by Artifactory: `xxxxx-0.0.0_327_gaad4634-cp36-cp36m-linux_x86_64.whl`

Briefly, I successfully uploaded the Artifact through a Concourse `put` step. The subsequent `get` step that is always performed by Concourse, however, failed because the version string is unrecognized.

I verified that the package indeed existed on our private PyPI after the job failed. I expected that the `put` should step should have failed if the version string was invalid.

I was told by our infra team that the resource version that we are using is: `0.5.2`. I am not able to easily test another version.

### Pipeline configuration

```
- put: release-whl.pypi
    inputs: [release-build]
    params:
      glob: release-build/*/xxxxx-*-linux_x86_64.whl
      wait_for_availability: true
      availability_waiting_time: 60
      max_availability_check_retry: 30
```

#### Put
```
Uploading distributions to https://xxxxx
Uploading xxxxx-0.0.0_327_gaad4634-cp36-cp36m-linux_x86_64.whl
100%|██████████| 151M/151M [00:04<00:00, 34.7MB/s] 
Finding package to upload
Glob release-build/*/xxxxx-*-linux_x86_64.whl matched files: ['/tmp/build/put/release-build/0.0.0-327-gaad4634/xxxxx-0.0.0_327_gaad4634-cp36-cp36m-linux_x86_64.whl']
Uploading /tmp/build/put/release-build/0.0.0-327-gaad4634/xxxxx-0.0.0_327_gaad4634-cp36-cp36m-linux_x86_64.whl version 0.0.0-327-gaad4634
Try 1 of 30 to call function `check_availability` from module `pypi_resource.out`
```

#### Get

```
Output directory: /tmp/build/get
Traceback (most recent call last):
  File "/opt/resource/in", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/pypi_resource/in_.py", line 102, in main
    response = in_(destdir, sys.stdin)
  File "/usr/local/lib/python3.6/site-packages/pypi_resource/in_.py", line 75, in in_
    common.merge_defaults(resconfig)
  File "/usr/local/lib/python3.6/site-packages/pypi_resource/common.py", line 173, in merge_defaults
    resconfig['version']['version'] = pipio.Version(resconfig['version']['version'])
  File "/usr/local/lib/python3.6/site-packages/pip/_vendor/packaging/version.py", line 227, in __init__
    raise InvalidVersion("Invalid version: '{0}'".format(version))
pip._vendor.packaging.version.InvalidVersion: Invalid version: '0.0.0-327-gaad4634'
```

**Comment by mboldt** [comment]
Thank you for reporting this. Do you have a preference on (1) forcing a more strict version format on `put` versus (2) allowing a looser version format on `get`? (My instinct is to go with (2) since the looser version `put` works today, but curious what you think.)

**Comment by kmdouglass** [comment]
> Thank you for reporting this. Do you have a preference on (1) forcing a more strict version format on `put` versus (2) allowing a looser version format on `get`? (My instinct is to go with (2) since the looser version `put` works today, but curious what you think.)

I do not have a strong preference, but I agree with you that option 2 (being less strict with the version on a `get` step) might be better. My reasoning is that the backend server (Artifactory, etc.) can be more strictly configured if necessary, and that this resource can defer responsibility of determining whether a version string is valid to the server.

**Comment by aliculPix4D** [comment]
Unfortunately, fixing the issue on PyPi server side (Artifactory in this case) is not at all trivial. Hence, forcing a more strict version format on put step could be a more practical solution. Additionally, since we can only fetch (get) the artifacts that are PEP440 compliant (independent of the pypi server),  forcing the same on the put step would make the resource more consistent.

I volunteer to open the PR.

**Comment by mboldt** [comment]
Considering this fixed with #37 merged.

**Comment by mboldt** [comment]
This is now released in 0.7.0.
